### PR TITLE
*: extract interface to be implemented

### PR DESCRIPTION
**Pull request by mahjonp**
### What is changed and how does it work?

- [x] Extract cluster `provisioner` interface as the adapter of cluster manager on K8s.
- ~~[ ] Refactor the `Nemesis` interface~~

To retain the backward compatibility, I extract out a `Node` struct and denotes a Pod on K8s environment.

That's means we cannot inject pod-kill kind faults using chaos-mesh currently, because they destroy the Pod and make the related `Node` be illegal. we need resolve this limitation in the future.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

Code changes

 - Has Go code change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: cmd/rawkv/main.go [hunk 1]
```diff
@@ -3,6 +3,7 @@ package main
 import (
 	"context"
 	"flag"
+	"github.com/pingcap/tipocket/pkg/cluster"
```
Comment (mapleFU) [inline cmd/rawkv/main.go]: group imports

File: cmd/tidb/main.go [hunk 2]
```diff
@@ -46,12 +46,12 @@ func main() {
 	switch *clientCase {
 	case "bank":
```
Comment (mapleFU) [inline cmd/tidb/main.go]: So, in m1 we only need to support `bank` and `long_fork`?

File: cmd/util/suit.go [hunk 3]
```diff
@@ -2,7 +2,7 @@ package util
 
 import (
 	"context"
-	"fmt"
+	"github.com/pingcap/tipocket/pkg/cluster"
```
Comment (mapleFU) [inline cmd/util/suit.go]: ditto

File: cmd/util/suit.go [hunk 4]
```diff
@@ -18,6 +18,8 @@ import (
 // Suit is a basic chaos testing suit with configurations to run chaos.
 type Suit struct {
 	*control.Config
+	// Provisioner deploy the SUT cluster
```
Comment (mapleFU) [inline cmd/util/suit.go]: What do you mean by "SUT"?

File: db/cluster/cluster.go [hunk 5]
```diff
@@ -3,6 +3,7 @@ package cluster
 import (
 	"context"
 	"fmt"
+	"github.com/pingcap/tipocket/pkg/cluster"
```
Comment (mapleFU) [inline db/cluster/cluster.go]: ditto

File: pkg/cluster/cluster.go [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+	PodName string
+
+	IP   string
+	Port string
+}
+
+// Provisioner provides a collection of APIs to deploy/destroy a cluster
+type Provisioner interface {
+	// SetUp sets up cluster, returns err or all nodes info
+	SetUp(ctx context.Context, spec json.RawMessage) (error, []Node)
```
Comment (you06) [inline pkg/cluster/cluster.go]: Will this interface change in the future, compare with RawMessage input, I prefer a struct or an interface.

Comment (mahjonp) [reply pkg/cluster/cluster.go]: yep, you're right. I'll change it.

File: cmd/util/suit.go (hunk 4 above)
Comment (mahjonp) [reply cmd/util/suit.go]: SUT means `system under test`.

File: cmd/tidb/main.go (hunk 2 above)
Comment (mahjonp) [reply cmd/tidb/main.go]: yeah.

**Pull request merged**